File: 엄마호랑이반/9주차 문제/이다영/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int cnt = 0;
		
	while (storey > 0) {
		int last = storey % 10;
	
		if (last >= 6 || (last == 5 && (storey / 10) % 10 >= 5)) {
			storey += 10 - last;
			cnt += 10 - last;
			
		} else {
			storey -= last;
			cnt += last;
		}
	
		storey /= 10;
	}
	
	return cnt;
    }
}

File: 엄마호랑이반/9주차 문제/최혜령/마법의엘리베이터.py
def solution(storey):
    answer = 0
    storey = abs(storey) # 절댓값으로 변환 (양수만 계산)
    while storey:
        print(f"current storey:{storey} answer:{answer}")
        div=storey%10
        if (div==5 and storey//10%10>=5) or div>5:
            storey+=10-div
            answer+=10-div
        else:
            answer+=div
        storey=storey//10
        print(f"end storey:{storey} answer:{answer}")
        
    return answer
